src/cli/util.js

```diff
@@ -535,10 +533,38 @@ function formatFiles(context) {
       } else {
         process.exitCode = 2;
       }
-    } else if (!context.argv["list-different"]) {
+    } else if (!context.argv["check"] && !context.argv["list-different"]) {
       writeOutput(context, result, options);
     }
+
+    if (
+      (context.argv["check"] || context.argv["list-different"]) &&
+      isDifferent
+    ) {
+      context.logger.log(filename);
+      process.exitCode = 1;
+    }
   });
+
+  // Print check summary based on expected exit code
+  if (context.argv["check"]) {
+    context.logger.log(
+      process.exitCode !== 1
+        ? "All matched files use Prettier code style!"
+        : context.argv["write"]
+        ? "Code style issues fixed in the above file(s)."
+        : "Code style issues found in the above file(s). Forgot to run Prettier?"
+    );
+  }
+
+  // Reset exitCode to 0 when using check/list-different is combined with write
+  if (
+    (context.argv["check"] || context.argv["list-different"]) &&
+    context.argv["write"] &&
+    process.exitCode === 1
+  ) {
+    process.exitCode = 0;
```

Comment:
Using `process.exitCode` to identify if it's formatted does not look like a good idea (especially we did not define what's the meaning of every exit code, and it's not a single-way mutation), can you change it to use a non-global variable?

Reply:
Hi @ikatyang 👋 OK, will do. Just curious: how many days / weeks / months do I roughly have before 1.16 doors a closing? Need to plan stuff or the NY break.

Reply:
At least 3 weeks I guess. (We'll spend probably a week writing/reviewing the blog post so you'll get the notification one week before the release.)

Reply:
@ikatyang this should be done now (I actually managed to implement a quick patch during lunch 🖥🍴) WDYT?

BTW: Thanks to adding `numberOfUnformattedFilesFound`, it is now possible to write  
  `Code style issues found in the above file. Forgot to run Prettier?`  
  `Code style issues found in the above files. Forgot to run Prettier?`  
instead of  
  `Code style issues found in the above file(s). Forgot to run Prettier?`  

Despite that this would look nicer and would not require much effort to implement, I've still decided to keep things as is. The reason is that there is a chance that someone might copy the plural form into another tool and check only against it. This can work fine in tests, but may fail in some real word cases. Prettier's CLI output is facing developers and developer tools, not the end-users, so this linguistic compromise looks justified to me. Let me know if you have a different opinion 😉

Reply:
IMO the `--check` option is more for end-users rather than tooling.

Reply:
I agree that `--check` is designed more for users than for the tools, but we can't be 100% certain. Having `(s)` in the end of a word is not that uncommon for messages that are coming from various dev tools, so I don't see that making the language look 100% fluent is worth the effort and risks. If we wanted to go for it, we'd need extra tests and probably an extra example in docs, but this would not add any measurable value to the end users after all.

I could find `(s)` in a [recent blog post](https://github.com/prettier/prettier/blame/master/website/blog/2018-11-07-1.15.0.md#L332), where I'd expect the language bar to be higher. If that was considered acceptable, I believe we can apply the same trick in stdout too 😉